src/ssh_cmd.rs

```diff
@@ -0,0 +1,36 @@
+use anyhow::Result;
+use std::process::Command;
+
+pub fn has_user_socket(host: &str) -> Result<bool> {
+    // Get the output of `ssh -G <host>` this will have a standard
+    // lowercase represesntation of:
+    //
+    // <key> <value>
+    //
+    // so a basic match should be enough.
+    let output = Command::new("ssh").arg("-G").arg(host).output()?;
```

Comment:
nit: `.arg("--")` before host, otherwise if they manage to pass a host starting with a hyphen, that'll get interpreted as ssh flags.